### PR TITLE
switchdev: Add GetVfRepresentor API

### PR DESCRIPTION
**Pull request by moshe010**
Signed-off-by: Moshe Levi <moshele@mellanox.com>

### CODE REVIEW

File: sriovnet_helper.go [hunk 1]
```diff
@@ -9,6 +9,7 @@ import (
 
 const (
 	NetSysDir        = "/sys/class/net"
+	PciSysDir        = "/sys/bus/pci/devices/"
```
Comment (paravmellanox) [inline sriovnet_helper.go]: no trailing /

File: sriovnet_switchdev.go [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	netdevPhysPortName = "phys_port_name"
+)
+
+var physPortRe = regexp.MustCompile(`pf(\d+)vf(\d+)`)
+
+func parsePortName(physPortName string) (pfRepIndex int, vfRepIndex int, err error) {
+
+	pfRepIndex = -1
+	vfRepIndex = -1
+	err = nil
```
Comment (paravmellanox) [inline sriovnet_switchdev.go]: no need to do err = nil

File: sriovnet_helper.go (hunk 1 above)
Comment (moshe010) [reply sriovnet_helper.go]: sorry it not belong to this patch I will remove it

Comment (paravmellanox) [reply sriovnet_helper.go]: ok

File: sriovnet_switchdev.go [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+			if err == nil {
+				vfRepIndex, err = strconv.Atoi(matches[2])
+			}
+		}
+
+	}
+	return pfRepIndex, vfRepIndex, err
+}
+
+func GetVfRepresentor(uplink string, vfIndex int) (string, error) {
```
Comment (paravmellanox) [inline sriovnet_switchdev.go]: needs comment to explain what is 'uplink' such as uplink netdev. Returns Vf representor netdev.
Lets improve next time.

**Pull request merged**